doc/source/data/performance-tips.rst

```diff
@@ -3,105 +3,40 @@
 Performance Tips and Tuning
 ===========================
 
-Monitoring your application
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-View the Ray dashboard to monitor your application and troubleshoot issues. To learn
-more about the Ray dashboard, read :ref:`Ray Dashboard <observability-getting-started>`.
-
-Debugging Statistics
-~~~~~~~~~~~~~~~~~~~~
-
-You can view debug stats for your Dataset executions via :meth:`ds.stats() <ray.data.Dataset.stats>`.
-These stats can be used to understand the performance of your Dataset workload and can help you debug problematic bottlenecks. Note that both execution and iterator statistics are available:
-
-.. testcode::
+Optimizing transforms
+---------------------
 
-    import ray
-    import time
-
-    def pause(x):
-        time.sleep(.0001)
-        return x
-
-    ds = (
-        ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
-        .map(lambda x: x)
-        .map(pause)
-    )
-
-    for batch in ds.iter_batches():
-        pass
-
-    print(ds.stats())
-
-.. testoutput::
-    :options: +MOCK
-
-    Stage 1 ReadRange->Map->Map: 16/16 blocks executed in 0.37s
-    * Remote wall time: 101.55ms min, 331.39ms max, 135.24ms mean, 2.16s total
-    * Remote cpu time: 7.42ms min, 15.88ms max, 11.01ms mean, 176.15ms total
-    * Peak heap memory usage (MiB): 157.18 min, 157.73 max, 157 mean
-    * Output num rows: 625 min, 625 max, 625 mean, 10000 total
-    * Output size bytes: 3658 min, 4392 max, 4321 mean, 69150 total
-    * Tasks per node: 16 min, 16 max, 16 mean; 1 nodes used
-    * Extra metrics: {'obj_store_mem_alloc': 3658, 'obj_store_mem_freed': 5000, 'obj_store_mem_peak': 40000}
-
-    Dataset iterator time breakdown:
-    * Total time user code is blocked: 551.67ms
-    * Total time in user code: 144.97us
-    * Total time overall: 1.01s
-    * Num blocks local: 0
-    * Num blocks remote: 0
-    * Num blocks unknown location: 16
-    * Batch iteration time breakdown (summed across prefetch threads):
-        * In ray.get(): 75.68us min, 220.26us max, 131.89us avg, 2.11ms total
-        * In batch creation: 326.58us min, 1.37ms max, 644.86us avg, 25.79ms total
-        * In batch formatting: 101.81us min, 898.73us max, 172.38us avg, 6.9ms total
-
-Batching Transforms
+Batching transforms
 ~~~~~~~~~~~~~~~~~~~
 
 Mapping individual records using :meth:`.map(fn) <ray.data.Dataset.map>` can be quite slow.
-Instead, consider using :meth:`.map_batches(batch_fn, batch_format="pandas") <ray.data.Dataset.map_batches>` and writing your ``batch_fn`` to
-perform vectorized pandas operations.
-
-.. _data_format_overheads:
+Instead, consider using :meth:`.map_batches(batch_fn) <ray.data.Dataset.map_batches>` and writing your ``batch_fn`` to
+perform vectorized operations.
```

Comment:
I think we want to clarify this more, otherwise it ends up with the same confusion as the warning.

I think let's explicitly say there is no performance benefit if your UDF is not vectorized.